Ja, komt later :) kan mss nog definieren dat als expliciet -1 is dat als uniform wordt beschouwd en indien iets anders negatief dan illegal keyword.
Ben momenteel wat aan het prutsen met mergen van master branch etc. en en dan te doen compileren. (momenteel compileert de master branch niet op lvserver: ``checking for gsl... configure: error: Package requirements (gsl) were not met:
No package 'gsl' found``)

--- a/src/polycap-photon.c
+++ b/src/polycap-photon.c
@@ -256,7 +256,7 @@ int polycap_photon_launch(polycap_photon *photon, polycap_error **error)
 	polycap_photon_scatf(photon, error);
 
 	//define polycapillary-to-photonsource axis 
-	//!!NOTE:this has to be changed. Now we assume all sources are in a straight line with PC central axis!!
+	//!!TODO:this has to be changed. Now we assume all sources are in a straight line with PC central axis!!
 	//e.g. no PC tilt
 	central_axis.x = 0;
 	central_axis.y = 0;

--- a/src/polycap-source.c
+++ b/src/polycap-source.c
@@ -77,7 +77,9 @@ polycap_photon* polycap_source_get_photon(polycap_source *source, polycap_rng *r
 	src_start_coords.x = src_start_x;
 	src_start_coords.y = src_start_y;
 	src_start_coords.z = 0;
-	if((source->src_sigx * source->src_sigy) < 1.e-20){ //uniform distribution over PC entrance
+	// if source->src_sigx or source->src_sigy are negative, assume uniform distribution over PC entrance
+	// otherwise, photon direction vector is within +- sigx or sigy
+	if( source->src_sigx < 0. || source->src_sigy < 0.){ //uniform distribution over PC entrance
 		start_direction.x = start_coords.x - src_start_x;
 		start_direction.y = start_coords.y - src_start_y;
 		start_direction.z = source->d_source;

--- a/src/polycap-source.c
+++ b/src/polycap-source.c
@@ -135,14 +135,6 @@ polycap_source* polycap_source_new(polycap_description *description, double d_so
 		polycap_set_error_literal(error, POLYCAP_ERROR_INVALID_ARGUMENT, "polycap_source_new: src_y must be greater than 0");
 		return NULL;
 	}
-	if (&src_sigx == NULL) {
-		polycap_set_error_literal(error, POLYCAP_ERROR_INVALID_ARGUMENT, "polycap_source_new: src_sigx cannot be NULL");
-		return NULL;
-	}
-	if (&src_sigy == NULL) {
-		polycap_set_error_literal(error, POLYCAP_ERROR_INVALID_ARGUMENT, "polycap_source_new: src_sigy cannot be NULL");
-		return NULL;
-	}
 
 	source = malloc(sizeof(polycap_source));
 	if(source == NULL){
@@ -283,16 +275,6 @@ polycap_source* polycap_source_new_from_file(const char *filename, polycap_error
 		polycap_source_free(source);
 		return NULL;
 	}
-//	if (source->src_sigx < 0.0){
-//		polycap_set_error_literal(error, POLYCAP_ERROR_INVALID_ARGUMENT, "polycap_source_new_from_file: source_temp->src_sigx must be greater than 0.0");
-//		polycap_source_free(source);
-//		return NULL;
-//	}
-//	if (source->src_sigy < 0.0){
-//		polycap_set_error_literal(error, POLYCAP_ERROR_INVALID_ARGUMENT, "polycap_source_new_from_file: source_temp->src_sigy must be greater than 0.0");
-//		polycap_source_free(source);
-//		return NULL;
-//	}
 	if (description->n_cap < 1){
 		polycap_set_error_literal(error, POLYCAP_ERROR_INVALID_ARGUMENT, "polycap_source_new_from_file: description->n_cap must be greater than 1");
 		polycap_source_free(source);

--- a/src/polycap-source.c
+++ b/src/polycap-source.c
@@ -11,7 +11,7 @@ polycap_photon* polycap_source_get_photon(polycap_source *source, polycap_rng *r
 	double n_shells; //amount of capillary shells in polycapilary
 	polycap_vector3 start_coords, start_direction, start_electric_vector, src_start_coords;
 	double r; //random number
-	int boundary_check;
+	int boundary_check = 0;
 	double src_rad, phi; //distance from source centre along angle phi from x axis
 	double src_start_x, src_start_y;
 	polycap_photon *photon;
@@ -46,26 +46,6 @@ polycap_photon* polycap_source_get_photon(polycap_source *source, polycap_rng *r
 		}
 	}
 
-	// Obtain photon start coordinates
-	n_shells = round(sqrt(12. * description->n_cap - 3.)/6.-0.5);
-	if(n_shells == 0.){ //monocapillary case
-		r = polycap_rng_uniform(rng);
-		start_coords.x = (2.*r-1.) * description->profile->cap[0];
-		r = polycap_rng_uniform(rng);
-		start_coords.y = (2.*r-1.) * description->profile->cap[0];
-		start_coords.z = 0.;
-	} else { // polycapillary case
-		// select random coordinates, check whether they are inside polycap boundary
-		boundary_check = 0;
-		do{
-			r = polycap_rng_uniform(rng);
-			start_coords.x = (2.*r-1.) * description->profile->ext[0];
-			r = polycap_rng_uniform(rng);
-			start_coords.y = (2.*r-1.) * description->profile->ext[0];
-			start_coords.z = 0.;
-			boundary_check = polycap_photon_within_pc_boundary(description->profile->ext[0], start_coords, error);
-		} while(boundary_check == 0);
-	}
 
 	// Obtain point from source as photon origin, determining photon start_direction
 	r = polycap_rng_uniform(rng);
@@ -80,16 +60,41 @@ polycap_photon* polycap_source_get_photon(polycap_source *source, polycap_rng *r
 	// if source->src_sigx or source->src_sigy are negative, assume uniform distribution over PC entrance
 	// otherwise, photon direction vector is within +- sigx or sigy
 	if( source->src_sigx < 0. || source->src_sigy < 0.){ //uniform distribution over PC entrance
+		// Obtain photon start coordinates
+		n_shells = round(sqrt(12. * description->n_cap - 3.)/6.-0.5);
+		if(n_shells == 0.){ //monocapillary case
+			r = polycap_rng_uniform(rng);
+			start_coords.x = (2.*r-1.) * description->profile->cap[0];
+			r = polycap_rng_uniform(rng);
+			start_coords.y = (2.*r-1.) * description->profile->cap[0];
+		} else { // polycapillary case
+			// select random coordinates, check whether they are inside polycap boundary
+			boundary_check = 0;
+			do{
+				r = polycap_rng_uniform(rng);
+				start_coords.x = (2.*r-1.) * description->profile->ext[0];
+				r = polycap_rng_uniform(rng);
+				start_coords.y = (2.*r-1.) * description->profile->ext[0];
+				boundary_check = polycap_photon_within_pc_boundary(description->profile->ext[0], start_coords, error);
+			} while(boundary_check == 0);
+		}
+		//now determine direction photon must have had in order to bridge src_start_coords and start_coords
 		start_direction.x = start_coords.x - src_start_x;
 		start_direction.y = start_coords.y - src_start_y;
 		start_direction.z = source->d_source;
 	} else { //non-uniform distribution, direction vector is within +- sigx
+		//first determine random photon direction
 		r = polycap_rng_uniform(rng);
 		start_direction.x = source->src_sigx * (1.-2.*fabs(r));
 		r = polycap_rng_uniform(rng);
 		start_direction.y = source->src_sigy * (1.-2.*fabs(r));
 		start_direction.z = 1.;
+		//now propagate photon towards polycap at position source->d_source
+		//This has a significant likelyhood the photons will miss the polycap and are thus not transmitted!!
+		start_coords.x = src_start_coords.x + start_direction.x * source->d_source;
+		start_coords.y = src_start_coords.y + start_direction.y * source->d_source;
 	}
+	start_coords.z = 0.;
 	polycap_norm(&start_direction);
 
 	// Provide random electric vector
@@ -130,12 +135,12 @@ polycap_source* polycap_source_new(polycap_description *description, double d_so
 		polycap_set_error_literal(error, POLYCAP_ERROR_INVALID_ARGUMENT, "polycap_source_new: src_y must be greater than 0");
 		return NULL;
 	}
-	if (src_sigx < 0.) {
-		polycap_set_error_literal(error, POLYCAP_ERROR_INVALID_ARGUMENT, "polycap_source_new: src_sigx must be greater than 0");
+	if (&src_sigx == NULL) {
+		polycap_set_error_literal(error, POLYCAP_ERROR_INVALID_ARGUMENT, "polycap_source_new: src_sigx cannot be NULL");
 		return NULL;
 	}
-	if (src_sigy < 0.) {
-		polycap_set_error_literal(error, POLYCAP_ERROR_INVALID_ARGUMENT, "polycap_source_new: src_sigy must be greater than 0");
+	if (&src_sigy == NULL) {
+		polycap_set_error_literal(error, POLYCAP_ERROR_INVALID_ARGUMENT, "polycap_source_new: src_sigy cannot be NULL");
 		return NULL;
 	}
 
@@ -278,16 +283,16 @@ polycap_source* polycap_source_new_from_file(const char *filename, polycap_error
 		polycap_source_free(source);
 		return NULL;
 	}
-	if (source->src_sigx < 0.0){
-		polycap_set_error_literal(error, POLYCAP_ERROR_INVALID_ARGUMENT, "polycap_source_new_from_file: source_temp->src_sigx must be greater than 0.0");
-		polycap_source_free(source);
-		return NULL;
-	}
-	if (source->src_sigy < 0.0){
-		polycap_set_error_literal(error, POLYCAP_ERROR_INVALID_ARGUMENT, "polycap_source_new_from_file: source_temp->src_sigy must be greater than 0.0");
-		polycap_source_free(source);
-		return NULL;
-	}
+//	if (source->src_sigx < 0.0){
+//		polycap_set_error_literal(error, POLYCAP_ERROR_INVALID_ARGUMENT, "polycap_source_new_from_file: source_temp->src_sigx must be greater than 0.0");
+//		polycap_source_free(source);
+//		return NULL;
+//	}
+//	if (source->src_sigy < 0.0){
+//		polycap_set_error_literal(error, POLYCAP_ERROR_INVALID_ARGUMENT, "polycap_source_new_from_file: source_temp->src_sigy must be greater than 0.0");
+//		polycap_source_free(source);
+//		return NULL;
+//	}
 	if (description->n_cap < 1){
 		polycap_set_error_literal(error, POLYCAP_ERROR_INVALID_ARGUMENT, "polycap_source_new_from_file: description->n_cap must be greater than 1");
 		polycap_source_free(source);

